### ISSUE TITLE
Numpy v1.26.4 build fails: Module "features" does not exist

### ISSUE THREAD

**Issue by hoodmane** [pyodide/pyodide#4498]
@ryanking13 @mattip @seberg

v1.26.3 build succeeds

Failure for v1.26.4:

<details><summary>Details</summary>
<p>

```
The Meson build system
Version: 1.3.1
Source dir: /home/hood/Documents/programming/pyodide/packages/numpy/build/numpy-1.26.4
Build dir: /home/hood/Documents/programming/pyodide/packages/numpy/build/numpy-1.26.4/.mesonpy-c6ekdglk
Build type: cross build
Project name: NumPy
Project version: 1.26.4
Cross compiler sanity tests disabled via the cross file.
C compiler for the host machine: /tmp/tmpqfmd_k4r/cc (emscripten 3.1.52 "emcc (Emscripten gcc/clang-like replacement + linker emulating GNU ld) 3.1.52 (fa478400df3351f7153c0279bc638784d3d90334)")
C linker for the host machine: /tmp/tmpqfmd_k4r/cc ld.wasm 18.0.0
C++ compiler for the host machine: /tmp/tmpqfmd_k4r/c++ (emscripten 3.1.52 "emcc (Emscripten gcc/clang-like replacement + linker emulating GNU ld) 3.1.52 (fa478400df3351f7153c0279bc638784d3d90334)")
C++ linker for the host machine: /tmp/tmpqfmd_k4r/c++ ld.wasm 18.0.0
Cython compiler for the host machine: cython (cython 3.0.8)
C compiler for the build machine: ccache cc (gcc 9.4.0 "cc (Ubuntu 9.4.0-1ubuntu1~20.04.2) 9.4.0")
C linker for the build machine: cc ld.bfd 2.34
C++ compiler for the build machine: ccache c++ (gcc 9.4.0 "c++ (Ubuntu 9.4.0-1ubuntu1~20.04.2) 9.4.0")
C++ linker for the build machine: c++ ld.bfd 2.34
Cython compiler for the build machine: cython (cython 3.0.8)
Build machine cpu family: x86_64
Build machine cpu: x86_64
Host machine cpu family: wasm32
Host machine cpu: wasm
Target machine cpu family: wasm32
Target machine cpu: wasm
Program python3 found: YES (/tmp/build-env-mr2vjazb/bin/python)
Found pkg-config: YES (/usr/bin/pkg-config) 0.29.1
Run-time dependency python found: YES 3.12
Has header "Python.h" with dependency python: YES 
Compiler for C supports arguments -fno-strict-aliasing: YES 

../meson_cpu/x86/meson.build:2:15: ERROR: Module "features" does not exist
```

</p>
</details>

**Comment by hoodmane** [comment]
Seems like the most likely culprit is:
https://github.com/numpy/numpy/pull/25748

**Comment by ryanking13** [comment]
According to this issue, https://github.com/numpy/numpy/issues/24750, it seems like vendored meson is not picked up for some reason.

**Comment by ryanking13** [comment]
I have a culprit. Let me check it out.

**Comment by solomoncyj** [comment]
error still persists
```
meson setup build
The Meson build system
Version: 1.5.99
Source dir: /home/solomoncyj/Downloads/numpy
Build dir: /home/solomoncyj/Downloads/numpy/build
Build type: native build
Project name: NumPy
Project version: 2.2.0.dev0+git20240817.d60444f
C compiler for the host machine: /usr/lib64/ccache/clang (clang 18.1.6 "clang version 18.1.6 (Fedora 18.1.6-3.fc40)")
C linker for the host machine: /usr/lib64/ccache/clang ld.bfd 2.41-37
C++ compiler for the host machine: /usr/lib64/ccache/clang++ (clang 18.1.6 "clang version 18.1.6 (Fedora 18.1.6-3.fc40)")
C++ linker for the host machine: /usr/lib64/ccache/clang++ ld.bfd 2.41-37
Cython compiler for the host machine: cython (cython 3.1.0)
Host machine cpu family: x86_64
Host machine cpu: x86_64
Program python3 found: YES (/home/solomoncyj/.local/share/pipx/venvs/meson/bin/python)
Found pkg-config: YES (/usr/bin/pkg-config) 2.1.1
Run-time dependency python found: YES 3.12
Has header "Python.h" with dependency python-3.12: YES 
Compiler for C supports arguments -fno-strict-aliasing: YES 
Compiler for C supports arguments -ftrapping-math: YES 

meson_cpu/x86/meson.build:2:15: ERROR: Module "features" does not exist
```

**Comment by agriyakhetarpal** [comment]
Hi @solomoncyj, I think this might be the problem?

```
Program python3 found: YES (/home/solomoncyj/.local/share/pipx/venvs/meson/bin/python)
```

This seems to be a non-standard location for the Python executable to be found. Could you try this within a fresh venv so that it picks up the vendored Meson instead of the one you have in your `pipx` installation directory? If it doesn't work and you continue to face more issues, please feel free to open a new issue with more details and a reproduction script, and we would be glad to help out.

**Comment by solomoncyj** [comment]
> Hi @solomoncyj, I think this might be the problem?
> 
> ```
> Program python3 found: YES (/home/solomoncyj/.local/share/pipx/venvs/meson/bin/python)
> ```
> 
> This seems to be a non-standard location for the Python executable to be found. Could you try this within a fresh venv so that it picks up the vendored Meson instead of the one you have in your `pipx` installation directory? If it doesn't work and you continue to face more issues, please feel free to open a new issue with more details and a reproduction script, and we would be glad to help out.

menson is installed a venv, and i am running the `meson setup build` command outside  of it  so you are saying i should have meson outside the venv?
this is how i cam to this point
```
sudo dnf in python3 pip git clang clang++ ninja
export CC=$(which clang)
export CXX=$(which clang++)
pip install pipx
git clone https://github.com/cython/cython
cd cython
pipx install .
git clone https://github.com/mesonbuild/meson
cd ../meson
pipx install .
git clone https://github.com/numpy/numpy
cd ../numpy
meson setup build
```

**Comment by agriyakhetarpal** [comment]
> so you are saying i should have meson outside the venv?

Yes. NumPy uses a vendored version of Meson which includes the "features" module, which won't exist in the one you set up in your virtual environment. Could you uninstall that and try out the vendored Meson?

**Comment by rubyFeedback** [comment]
This is the second Google result for "features" coming up. Googling for this was super-hard.

I finally managed to resolve it thanks to LFS/BLFS:

https://www.linuxfromscratch.org/blfs/view/svn/general/python-modules.html#numpy

I had to build the various pre-requisites before numpy. Not sure why this became so complicated,
it used to be simpler, but I am glad that numpy is now working, so I write this, then move on. \o/
